### PR TITLE
Improve repository migration scripts

### PR DESCRIPTION
**Pull request by njohner**
We add the possibility to split the task syncing to a different step, as the migration itself is done on an OGDS that is not the productive one. For that we pickle the migrator object which knows which tasks will then have to be synced. This last step will be done in debug mode.

For https://4teamwork.atlassian.net/browse/CA-1267

### CODE REVIEW

File: opengever/maintenance/scripts/repository_migration.py [hunk 1]
```diff
@@ -1179,16 +1246,27 @@ def main():
...
+    analyser.export_to_excel()
+
+    analyser_path = os.path.join(options.output_directory, "analyser.pkl")
+    with open(analyser_path, "w") as outfile:
+        # some attributes cannot be pickled so we simply set them to None
+        analyser.catalog = None
+        analyser.formatter = None
+        analyser.reporoot = None
+        analyser._reference_repository_mapping = None
+        pickle.dump(analyser, outfile)
```
Comment (deiferni) [inline opengever/maintenance/scripts/repository_migration.py:1269]: So, the only thing that you are interested here is the list of UIDs, right? I would advise against pickling the object. IMO you don't gain anything by the pickle, on the contrary it seems to be more complicated at first glance. Wouldn't something simpler like a text file with one item per line be sufficient here? The text file is human readable/editable and does not have any (ok, i admit, theoretical here) potential security impacts.

File: opengever/maintenance/scripts/repository_migration.py [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
+       syncing are stored in the migrator object that is then pickled.
+  -n : dry-run.
+
+If task syncing was skipped, it can be later performed in debug mode:
+
+from opengever.maintenance import dm; dm()
+from opengever.maintenance.scripts.repository_migration import *
+import pickle
+import transaction
+with open(path/to/migrator.pkl, "r") as migrtor_file:
```
Comment (deiferni) [inline opengever/maintenance/scripts/repository_migration.py:20]: migr**a**tor?

**Review by deiferni — APPROVED** [review]
LGTM apart from the pickling. Maybe i did not spot all the use-cases here. ☎️  if you wish 😉 .

File: opengever/maintenance/scripts/repository_migration.py (hunk 1 above)
Comment (njohner) [reply opengever/maintenance/scripts/repository_migration.py]: Ok I now write some data that might be of interest to json files, as well of course as the list of task UIDs that we will need for syncing...

**Pull request merged**